docs/concepts/smart-contracts.md

```diff
@@ -56,69 +52,73 @@ pub struct HyleOutput {
+These fields define state transitions.
 
-- `initial_state`: Matches the onchain `state_digest` before the transaction. The `initial_state` must match the onchain `state_digest`. Otherwise, the state transition is invalid.
+- `initial_state`: must match the onchain `state_digest` before the transaction. If they don't match, the state transition is invalid.
 - `next_state`: Represents the new onchain `state_digest` after the transaction.
 
-Smart contracts can adapt the actual structure of this field. In the future, fees will depend in part on the size of the `state_digest`, so we encourage you to keep it small.
-
-#### Identity
+Future fees may depend on `state_digest` size, so keep it minimal.
 
-The `identity` field identifies the person who initiates a transaction.
+### Identity
 
-Hylé does not have a native signature type. Instead, it uses the `identity` field of the first proof in the transaction to identify the sender. This allows you to use any kind of identity verification.
+!!! info
+    Read [our identity documentation](./identity.md) for details.
 
-The identity is in two parts:
+Identity consists of:
 
 1. An address;
-1. The name of the contract that the proof was generate for.
+1. The name of the contract that the proof was generated for.
 
-For example, if a smart contract was registered onchain to verify Ethereum EOAs, the `identity` for them would look like `0x1234...5678.eth_eoa` where `eth_eoa` is the name of the contract and the first part matches a regular Ethereum address.
+!!! example
+    A contract verifying Ethereum EOAs might have an identity like `0x1234...5678.eth_eoa` where the first part matches a regular Ethereum address and `eth_eoa` is the name of the contract.
 
-**Note**: All proofs in a transaction must declare the same identity as the first transaction or an empty identity.
+### Blob index and blobs
 
-#### TX Hash
+Each blob transaction includes multiple blobs:
 
-The `tx_hash` field hashes transaction data, preventing replay attacks and providing a means for contracts to access this information.
+- `index` uniquely identifies a blob within a transaction.
+- `blobs` is a list of all blobs included in the transaction.
 
-The field is not validated by the protocol.
+### TX Hash
 
-#### Blob index and blobs
+`tx_hash` is a hash of transaction data.
 
-Since a blob transaction can include several blobs, the design includes:
+The protocol does not validate this field and `tx_hash` may be deprecated in later versions.
 
-- `pub index: BlobIndex`: uniquely identifies a specific blob within a transaction.
-- `pub blobs: Vec<u8>`: all blobs included in the transaction.
+### Success
 
-#### Success
-
-This boolean field indicates whether the proof is for a successful transaction or a failure. It can be useful to prove that a transaction is invalid.
+This boolean field indicates whether the proof is for a successful or failed transactions. It can be used to prove that a transaction is invalid.
```

Comment:
Note: if a proof returns a "success = false", the whole blob transaction is marked as failed. There is no need of generating proofs for the other blobs.